resources/sqlite/src/lib.rs

```diff
@@ -0,0 +1,171 @@
+
+mod conn_opts;
+pub use conn_opts::*;
+
+use sqlx::sqlite::SqliteConnectOptions;
+/// The [`sqlx::SqlitePool`](https://docs.rs/sqlx/latest/sqlx/type.SqlitePool.html) that is being returned to the user.
+///
+pub use sqlx::SqlitePool;
+
+/// Builder struct used to configure the database, e.g. `SQLite(opts = SQLiteConnOpts::new())`.
+#[derive(Serialize)]
+pub struct ShuttleSqlite {
+    opts: ShuttleSqliteConnOpts,
+}
+
+impl ShuttleSqlite {
+    pub fn opts(mut self, opts: ShuttleSqliteConnOpts) -> Self {
+        self.opts = opts;
+        self
+    }
+
+    pub fn in_memory(mut self, on: bool) -> Self {
+        self.opts.in_memory = on;
+        self
+    }
+}
+
+#[async_trait]
+impl ResourceBuilder<sqlx::SqlitePool> for ShuttleSqlite {
+    /// The type of resource this creates
+    const TYPE: Type = Type::EmbeddedDatabase;
+
+    /// The internal config being constructed by this builder. This will be used to find cached [Self::Output].
+    type Config = Self;
+
+    /// The output type used to build this resource later
+    type Output = ShuttleSqliteConnOpts;
+
+    /// Create a new instance of this resource builder
+    fn new() -> Self {
+        Self {
+            opts: ShuttleSqliteConnOpts::default(),
+        }
+    }
+
+    /// Get the internal config state of the builder
+    ///
+    /// If the exact same config was returned by a previous deployement that used this resource, then [Self::output()]
+    /// will not be called to get the builder output again. Rather the output state of the previous deployment
+    /// will be passed to [Self::build()].
+    fn config(&self) -> &Self::Config {
+        &self
+    }
+
+    /// Get the config output of this builder
+    ///
+    /// This method is where the actual resource provisioning should take place and is expected to take the longest. It
+    /// can at times even take minutes. That is why the output of this method is cached and calling this method can be
+    /// skipped as explained in [Self::config()].
+    async fn output(
+        mut self,
+        factory: &mut dyn Factory,
+    ) -> Result<Self::Output, shuttle_service::Error> {
+        // We construct an absolute path using `storage_path` to prevent user access to other parts of the file system.
+        let storage_path = factory.get_storage_path()?;
+        self.opts.storage_path = storage_path;
+        Ok(self.opts)
+    }
+
+    /// Build this resource from its config output
+    async fn build(build_data: &Self::Output) -> Result<sqlx::SqlitePool, shuttle_service::Error> {
+        // This should never fail if our `try_from` is implemented correctly, which is guaranteed by our tests.
+        let opts = SqliteConnectOptions::try_from(build_data)?;
+
+        let pool = sqlx::SqlitePool::connect_with(opts)
+            .await
+            .map_err(|e| shuttle_service::Error::Database(e.to_string()))?;
+
+        Ok(pool)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use assert_json_diff::assert_json_eq;
```

Comment:
[`pretty_assertions`](https://docs.rs/pretty_assertions) will reduce the need to convert it to json first

Reply:
So *that's* the name of the crate I couldn't rememberˆˆ